src/plugins/data_source/server/client/configure_client.ts

```diff
@@ -144,12 +138,18 @@ const getQueryClient = async (
     });
     credential = credentialProvider.credential;
     type = credentialProvider.type;
+
+    if (credential.service === undefined) {
+      credential = { ...credential, service: dataSourceAttr.auth.credentials?.service };
+    }
   }
 
+  const cacheKey = endpoint;
```

Comment:
For same endpoint, are there any possibilities that we may send request with different AuthTypes?

For Example:
Test endpoint connection with UsernamePassword and SigV4.

In that case, UsernamePassword client is likely to be reused by SigV4 request?

Reply:
That's a really good point to bring up. As you can see below, we have two caches - one for regular clients (no auth and basic auth)  and one for aws clients.

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/31e8481fd2c0bf185d34c1140724e9320f6a676c/src/plugins/data_source/server/client/client_pool.ts#L24-L32

And while adding and fetching clients from cache, based on type it looks at correct cache.
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/31e8481fd2c0bf185d34c1140724e9320f6a676c/src/plugins/data_source/server/client/client_pool.ts#L76-L87

So answer to your question, UsernamePassword client and SigV4 client will not remain in same cache.